tests/test_published_examples.py

```diff
@@ -1,37 +1,36 @@
 # Copyright (C) 2014 Anaconda, Inc
 # SPDX-License-Identifier: BSD-3-Clause
 import os
+import sys
 
 import pytest
 
-import sys
-
-from conda_build import api
+from conda.testing.integration import BIN_DIRECTORY
+from conda_build.api import build
 from conda_build.utils import check_call_env
-from .utils import metadata_dir, is_valid_dir
-
-published_examples = os.path.join(os.path.dirname(metadata_dir), 'published_code')
+from .utils import published_dir, is_valid_dir
 
 
 @pytest.mark.sanity
 def test_skeleton_pypi(testing_workdir):
     """published in docs at https://docs.conda.io/projects/conda-build/en/latest/user-guide/tutorials/build-pkgs-skeleton.html"""
-    conda_path = os.path.join(sys.prefix, 'Scripts' if sys.platform == 'win32' else 'bin', 'conda')
-    cmd = conda_path + ' skeleton pypi click'
-    check_call_env(cmd.split())
-    cmd = conda_path + ' build click'
-    check_call_env(cmd.split())
-
+    conda_path = os.path.join(sys.prefix, BIN_DIRECTORY, "conda")
 
-@pytest.fixture(params=[dirname for dirname in os.listdir(published_examples)
-                        if is_valid_dir(published_examples, dirname)])
-def recipe(request):
-    return os.path.join(published_examples, request.param)
+    check_call_env([conda_path, "skeleton", "pypi", "click"])
+    check_call_env([conda_path, "build", "click"])
 
 
-# This tests any of the folders in the test-recipes/published_code folder that don't start with _
 @pytest.mark.sanity
+@pytest.mark.parametrize(
+    "recipe",
+    [
```

Comment:
Really not a big fan of these chonky inline list comprehensions. Could we at least assign it to a module level variable so we can better express the semantic meaning of it? I know you added a comment, which is nice, but maybe instead of have the comment appear in the middle of the list comprehension, it could appear above the variable assignment.

I'm not going to be nit-picky and block on this though. Please let me know what you think.

Reply:
hehe and I am not a fan of assigning temporary variables that clutter the namespace, especially those that are only used once

I agree that the comment should be moved so I'll take another look